process/block/shardblock.go

```diff
@@ -1055,30 +1055,25 @@ func (sp *shardProcessor) CommitBlock(
 		"nonce", highestFinalBlockNonce,
 	)
 
-	lastBlockHeader := sp.blockChain.GetCurrentBlockHeader()
-
-	committedRootHash, err := sp.accountsDB[state.UserAccountsState].RootHash()
-	if err != nil {
-		return err
-	}
+	rootHash := sp.getLastExecutedRootHash(header)
 
-	err = sp.setCurrentBlockInfo(header, headerHash, committedRootHash)
+	err = sp.setCurrentBlockInfo(header, headerHash, rootHash)
 	if err != nil {
 		return err
 	}
+	sp.blockChain.SetCurrentBlockHeaderHash(headerHash)
 
-	rootHash := getLastExecutionResultsRootHash(header, committedRootHash)
 	lastExecutionResultHeader, err := sp.getLastExecutionResultHeader(header)
 	if err != nil {
 		return err
 	}
-	sp.blockChain.SetCurrentBlockHeaderHash(headerHash)
 
 	err = sp.onExecutedBlock(lastExecutionResultHeader, rootHash)
 	if err != nil {
 		return err
 	}
 
+	lastBlockHeader := sp.blockChain.GetCurrentBlockHeader()
```

Comment:
previously lastBlockHeader was the previous block, as it was read before setting in blockchain the new header. Now this is the header being committed, as it is read after being set.

in the index function this is expected to be the previous block.